cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -625,8 +627,13 @@ public ArtifactDetail write(final Id.Artifact artifactId,
       }
     }
 
-    Location fileDirectory =
-      namespacedLocationFactory.get(artifactId.getNamespace(), ARTIFACTS_PATH).append(artifactId.getName());
+    Location fileDirectory = null;
+    try {
+      fileDirectory = namespacedLocationFactory.get(artifactId.getNamespace(),
+                                                    ARTIFACTS_PATH).append(artifactId.getName());
+    } catch (NamespaceNotFoundException | UnauthenticatedException e) {
+      throw Throwables.propagate(e);
```

Comment:
The only caller to this method is from `ArtifactRepository` and in there the method actually has `throws Exception`. I think it's better to have this method just `thorws Exception` as well instead of wrapping it.